movebucket_test.go

```diff
@@ -0,0 +1,482 @@
+					t.Fatalf("error dumping bucket %v to file %v", movedBucket.String(), dumpBucketAfterMoving)
+				}
+
+				beforeBucket := readBucketFromFile(t, dumpBucketBeforeMoving)
+				afterBucket := readBucketFromFile(t, dumpBucketAfterMoving)
+
+				if !bytes.Equal(beforeBucket, afterBucket) {
+					t.Fatalf("bucket's content before moving is different than after moving")
+				}
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+
+			t.Cleanup(func() {
+				os.RemoveAll(dumpBucketBeforeMoving)
+				os.RemoveAll(dumpBucketAfterMoving)
+			})
+		})
+	}
+}
+
+func createBucketIfNotExist(t testing.TB, tx *bbolt.Tx, paths ...string) *bbolt.Bucket {
+	t.Helper()
+
+	var bk *bbolt.Bucket
+	var err error
+
+	// skip if root bucket (i.e. bucket name == "")
+	if paths[0] != "" {
+		bk, err = tx.CreateBucketIfNotExists([]byte(paths[0]))
+		if err != nil {
+			t.Fatalf("error creating bucket %v: %v", paths[0], err)
+		}
+		insertRandKeysValuesBucket(t, bk, rand.Intn(4096))
+	}
+
+	for _, key := range paths[1:] {
+		if bk == nil {
+			bk, err = tx.CreateBucketIfNotExists([]byte(key))
+		} else {
+			bk, err = bk.CreateBucketIfNotExists([]byte(key))
+		}
+		if err != nil {
+			t.Fatalf("error creating bucket %v: %v", key, err)
+		}
+		insertRandKeysValuesBucket(t, bk, rand.Intn(4096))
+	}
+
+	return bk
+}
+
+func retrieveParentBucket(t testing.TB, tx *bbolt.Tx, paths ...string) *bbolt.Bucket {
+	paths = paths[:len(paths)-1]
+	return retrieveChildBucket(t, tx, paths...)
+}
+
+func retrieveChildBucket(t testing.TB, tx *bbolt.Tx, paths ...string) *bbolt.Bucket {
+	t.Helper()
+
+	var bk *bbolt.Bucket = nil
+	for _, path := range paths {
+		// skip root bucket
+		if path == "" {
+			continue
+		}
+		if bk == nil {
+			bk = tx.Bucket([]byte(path))
+		} else {
+			bk = bk.Bucket([]byte(path))
+		}
+		if bk == nil {
+			t.Fatalf("error retrieving bucket %v within paths %v", path, strings.TrimSuffix(strings.Join(paths, "->"), "->"))
+		}
+	}
+
+	return bk
+}
+
+func readBucketFromFile(t testing.TB, tmpFile string) []byte {
+	data, err := os.ReadFile(tmpFile)
+	if err != nil {
+		t.Fatalf("error reading temp file %v", tmpFile)
+	}
+
+	return data
+}
+
+func insertRandKeysValuesBucket(t testing.TB, bk *bbolt.Bucket, n int) {
+	var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	var min, max = 1, 1024
+
+	for i := 0; i < n; i++ {
+		// generate rand key/value length
+		keyLength := rand.Intn(max-min) + min
+		valLength := rand.Intn(max-min) + min
+
+		keyData := make([]rune, keyLength)
+		valData := make([]rune, valLength)
+
+		for j := range keyData {
+			keyData[j] = letters[rand.Intn(len(letters))]
+		}
+
+		for j := range valData {
+			valData[j] = letters[rand.Intn(len(letters))]
+		}
+
+		pErr := bk.Put([]byte(string(keyData)), []byte(string(valData)))
+		if pErr != nil {
+			t.Fatalf("error inserting key %v and value %v in bucket %v: %v", string(keyData), string(valData), bk.String(), pErr)
+		}
```

Comment:
Import "`crand "crypto/rand"`"
```suggestion
		 _, err := crand.Read(keyData)
		 require.NoError(t, err)

		_, err = crand.Read(valData)
		require.NoError(t, err)

		err = bk.Put(keyData, valData)
		require.NoError(t, err)
```

Reply:
done